plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala

```diff
@@ -97,14 +97,16 @@ trait AST { this: TransformCake ⇒
     if (m.isPublic) "public"
     else if (m.isProtected && !topLevel) "protected"
     else if (m.isPrivate && !topLevel) "private"
-    else "public" // this is the case for “private[xy]” and top level classes
+    else if (m.privateWithin != tpnme.EMPTY) ""
```

Comment:
I'm not very sure about how java doc works. Does it expect everything to type-check according to Java rules? If that's the case, I think this comment would apply everywhere Scala makes something public in classfiles that was private in source. So probably none of the changes here would be safe to do. AFAIK the compiler does not make anything public just for tha sake of it.